### DISCUSSION TITLE
Default trackers list

### DISCUSSION THREAD

**Discussion by depler** [transmission/transmission#4249] [General]
What is `default-trackers` settings in `settings.json` file? Is it really working or not?

**Comment by GaryElshaw** [comment]
I've never got my head around default trackers. Adding them, removing etc. As an idea, i always thought a blacklist was a good idea too. I always manually delete CopperSurfer, i have never been able to connect to that tracker in Transmission, to the point where i just immediately delete it.

**Comment by ckerr** [comment]
@depler it's for users who want to have a default set of trackers to announce _all_ their public torrents to.

It's working for me, anyway.

**Reply by depler** [reply]
I've tried to add some default trackers, but they didn't show in torrent trackers list (I am using transmission GUI to see trackers list). How exactly it works?

**Reply by GaryElshaw** [reply]
This came up recently, and Antoine can articulate it better than me. The .json file, in macOS, doesn't function as it does on other platforms? @Coeur

**Reply by GaryElshaw** [reply]
Probably irrelevant, but I see multiple instances of contrary information between the macOS client 'settings' and the .json file in 'Application Support'.

Begging the project question: Advanced macOS users look to the json file for granular preference handling. It would be great to discuss how this is handled so there is a better confluence between users and advanced users.

**Reply by ckerr** [reply]
> I am using transmission GUI to see trackers list

@depler which transmission GUI are you using?

**Reply by depler** [reply]
@ckerr this one: https://github.com/transmission-remote-gui/transgui

**Reply by ckerr** [reply]
@depler I like that app but it's not maintained by the Transmission org; IDK if it supports editing default trackers yet.

**Comment by MarkMain** [comment]
It would be more ideal for users to have the ability to modify the default tracker list on screen provided in the app.

It would also be nice if duplicates were removed when trying to add new trackers to that list or also to the list on a specific torrent.

**Comment by ckerr** [comment]
> It would be more ideal for users to have the ability to modify the default tracker list on screen provided in the app.

This is the case in `transmission-gtk` and `transmission-qt`; I don't remember if this is true in the macOS client or not. If it's not there, please open a bug report for it :+1: 

> It would also be nice if duplicates were removed when trying to add new trackers to that list or also to the list on a specific torrent.

In practice it works out OK since duplicates are removed internally when Transmission decides what trackers to announce to announce a torrent to. IIRC duplicates are allowed in the UI just for ease-of-use when editing, e.g. adding tracker X to some torrents before removing X from the default list.

**Comment by tollfree** [comment]
It seems that there is no such option in macOS client.
I found it only in Web interface.

**Reply by Coeur** [reply]
Kindly open an issue to advertise about this feature request, and one may implement it one day?
